content/cookbook/transactions/send-sol.md

```diff
@@ -8,6 +8,127 @@ description:
 
 To send SOL, you will need to interact with the [SystemProgram][1].
 
+<Tabs groupId="language" items={['web3.js v2', 'web3.js v1']}>
+
+<Tab value="web3.js v2">
+
+```typescript filename="send-sol.ts" {70-74}
+import {
+  address,
+  airdropFactory,
+  appendTransactionMessageInstructions,
+  createSolanaRpc,
+  createSolanaRpcSubscriptions,
+  createTransactionMessage,
+  devnet,
+  generateKeyPairSigner,
+  getComputeUnitEstimateForTransactionMessageFactory,
+  getSignatureFromTransaction,
+  lamports,
+  pipe,
+  prependTransactionMessageInstructions,
+  sendAndConfirmTransactionFactory,
+  setTransactionMessageFeePayerSigner,
+  setTransactionMessageLifetimeUsingBlockhash,
+  signTransactionMessageWithSigners,
+} from "@solana/web3.js";
+import {
+  getSetComputeUnitLimitInstruction,
+  getSetComputeUnitPriceInstruction,
+} from "@solana-program/compute-budget";
+import { getAddMemoInstruction } from "@solana-program/memo";
+import { getTransferSolInstruction } from "@solana-program/system";
+
+async function transferSol() {
+  // Create an RPC. Use localnet for solana-test-validator. This will get you easier airdrops.
+  const CLUSTER = "devnet";
+  const rpc = createSolanaRpc(devnet(`https://api.${CLUSTER}.solana.com`));
+  const rpcSubscriptions = createSolanaRpcSubscriptions(
+    devnet(`wss://api.${CLUSTER}.solana.com`),
+  );
+
+  // Create an airdrop function.
+  const airdrop = airdropFactory({ rpc, rpcSubscriptions });
+
+  // Create a utility that estimates a transaction message's compute consumption.
+  const getComputeUnitEstimate =
+    getComputeUnitEstimateForTransactionMessageFactory({ rpc });
+
+  // Create a transaction sending function.
+  const sendAndConfirmTransaction = sendAndConfirmTransactionFactory({
+    rpc,
+    rpcSubscriptions,
+  });
+
+  // Create and fund an account.
+  const keypairSigner = await generateKeyPairSigner();
+  console.log("Created an account with address", keypairSigner.address);
+  console.log("Requesting airdrop");
+  await airdrop({
+    commitment: "confirmed",
+    lamports: lamports(1000_000n),
+    recipientAddress: keypairSigner.address,
+  });
+  console.log("Airdrop confirmed");
+
+  // Create a memo transaction.
+  console.log("Creating a memo transaction");
+  const { value: latestBlockhash } = await rpc.getLatestBlockhash().send();
+  const transactionMessage = pipe(
+    createTransactionMessage({ version: "legacy" }),
+    m => setTransactionMessageFeePayerSigner(keypairSigner, m),
+    m => setTransactionMessageLifetimeUsingBlockhash(latestBlockhash, m),
+    m =>
+      appendTransactionMessageInstructions(
+        [
+          getSetComputeUnitPriceInstruction({ microLamports: 5000n }),
+          getTransferSolInstruction({
+            source: keypairSigner,
+            destination: address("web3Qm5PuFapMJqe6PWRWfRBarkeqE2ZC8Eew3zwHH2"),
+            amount: lamports(1_000n),
```

Comment:
When running locally, this transfer throws an error because 1000 lamports is less than minimum required for an account.


```
SolanaError: Transaction failed when it was simulated in order to estimate the compute unit consumption. The compute unit estimate provided is for a transaction that failed when simulated and may not be representative of the compute units this transaction would consume if successful. Inspect the `cause` property of this error to learn more
    at getComputeUnitEstimateForTransactionMessage_INTERNAL_ONLY_DO_NOT_EXPORT
    ...
  cause: { InsufficientFundsForRent: { account_index: 1n } },
  context: { __code: 5663019, unitsConsumed: 450 }
  ```